Sources/Lingo/DataSources/FileDataSource.swift

```diff
@@ -69,32 +59,30 @@ fileprivate extension FileDataSource {
     ///   "one": "You have an unread message."
     ///   "many": "You have %{count} unread messages."
     /// }
-    func pluralizedValues(fromRaw rawPluralizedValues: [String: String]) -> [PluralCategory: String] {
-        var pluralizedValues = [PluralCategory: String]()
+    func pluralizedValues(fromRaw rawPluralizedValues: [String: String]) throws -> [PluralCategory: String] {
+        var result = [PluralCategory: String]()
         
         for (rawPluralCategory, value) in rawPluralizedValues {
             guard let pluralCategory = PluralCategory(rawValue: rawPluralCategory) else {
-                assertionFailure("Unsupported plural category: \(rawPluralCategory)")
-                continue
+                throw Error.parsingFailure(message: "Unsupported plural category: \(rawPluralCategory)")
             }
-            pluralizedValues[pluralCategory] = value
+            result[pluralCategory] = value
         }
         
-        return pluralizedValues
+        return result
     }
     
     /// Loads a localizations file from disk if it exists and parses it.
-    func loadLocalizations(atPath path: String) -> [String: Any]? {
-        precondition(FileManager().fileExists(atPath: path))
-
-        guard
-            let fileContent = try? Data(contentsOf: URL(fileURLWithPath: path)),
-            let jsonObject = try? JSONSerialization.jsonObject(with: fileContent, options: []) as? [String: Any] else {
-                assertionFailure("Failed reading localizations from file at path: \(path)")
-                return nil
+    func loadLocalizations(atPath path: String) throws -> [String: Any] {
```

Comment:
the usage of this function looks like: `loadLocalizations(atPath: jsonFilePath)`
would it might be better like: `loadLocalizations(at: jsonFilePath)` ?

Reply:
I was inspired by NSFileManager:
```
open func contentsOfDirectory(atPath path: String) throws -> [String]
open func subpathsOfDirectory(atPath path: String) throws -> [String]
open func attributesOfItem(atPath path: String) throws -> [FileAttributeKey : Any]
```

so I think it is fine to keep it this way.